### ISSUE TITLE
Site-based config split not activating on ACSF

### ISSUE THREAD

**Issue by mikecrittenden** [acquia/blt#3161]
My system information:
* Operating system type: macOS
* Operating system version: Mojave
* BLT version: 9.2 alpha

<details><summary>Output of `blt doctor`:</summary>
<p>

```
+---------------------------+------------------------------------------------+
| Property                  | Value                                          |
+---------------------------+------------------------------------------------+
| %paths.%root              | /var/www/docroot                               |
| %paths.%site              | sites/default                                  |
| %paths.%modules           | sites/all/modules                              |
| %paths.%themes            | sites/all/themes                               |
| %paths.%config-sync       | /var/www/config/default                        |
| %paths.%files             | sites/default/files                            |
| %paths.%temp              | /tmp                                           |
| %paths.%private           | /var/www/files-private                         |
| admin-theme               | seven                                          |
| alias-searchpaths.0       | /var/www/drush/sites                           |
| blt-version               | 9.2.0-alpha3                                   |
| bootstrap                 | Successful                                     |
| composer-version          | Composer version 1.6.3 2018-01-31 16:28:17     |
| config-sync               | /var/www/config/default                        |
| db-driver                 | mysql                                          |
| db-hostname               | db                                             |
| db-name                   | fanniemae                                      |
| db-password               | root                                           |
| db-port                   | 3306                                           |
| db-status                 | Connected                                      |
| db-username               | root                                           |
| drupal-settings-file      | sites/default/settings.php                     |
| drupal-version            | 8.6.1                                          |
| drush-alias-files.0       | /var/www/drush/sites/default.site.yml          |
| drush-alias-files.1       | /var/www/drush/sites/fhl.site.yml              |
| drush-alias-files.2       | /var/www/drush/sites/kyo.site.yml              |
| drush-cache-directory     | /home/docker/.drush/cache                      |
| drush-conf.0              | /var/www/vendor/drush/drush/drush.yml          |
| drush-conf.1              | /var/www/drush/drush.yml                       |
| drush-conf.2              | /var/www/docroot/sites/default/local.drush.yml |
| drush-script              | /var/www/vendor/bin/drush                      |
| drush-temp                | /tmp                                           |
| drush-version             | 9.4.0                                          |
| files                     | sites/default/files                            |
| install-profile           | microsite                                      |
| modules                   | sites/all/modules                              |
| php-bin                   | /usr/local/bin/php                             |
| php-conf.1                | false                                          |
| php-os                    | Linux                                          |
| private                   | /var/www/files-private                         |
| root                      | /var/www/docroot                               |
| site                      | sites/default                                  |
| stacks.drupal-vm.inited   | false                                          |
| stacks.dev-desktop.inited | false                                          |
| temp                      | /tmp                                           |
| theme                     | particle                                       |
| themes                    | sites/all/themes                               |
| uri                       | http://fanniemae.docksal                       |
+---------------------------+------------------------------------------------+
+--------------------------------------+-----------------------------------------------------------+
| Check                                | Problem                                                   |
+--------------------------------------+-----------------------------------------------------------+
| NodeCheck:checkNodeVersionFileExists | Neither .nvmrc nor .node-version file found in repo root. |
+--------------------------------------+-----------------------------------------------------------+

```

</p>
</details>

I have a config split named `fhl` which lives at `config/default/config_split.config_split.fhl.yml`. Here are its contents:

```yaml
uuid: c72b7ff8-b36a-45b1-b464-7677d234e04a
langcode: en
status: false
dependencies: {  }
id: fhl
label: 'Site: FHL'
folder: ../config/sites/fhl
module: {  }
theme: {  }
blacklist: {  }
graylist:
  - system.site
graylist_dependents: true
graylist_skip_equal: false
weight: 0
```

This should map to the website which lives at `http://fhl.dev-OURSITE.acsitefactory.com`. But it doesn't appear to be working. When I visit the config split admin UI on that website, it shows that the "fhl" split is inactive. 

Looking at the BLT code, I see this in `config.settings.php`:

```php
// Set acsf site split if explicit global exists.
if (isset($acsf_site_name)) {
  $config["$split_filename_prefix.$acsf_site_name"]['status'] = TRUE;
}
```

And I can confirm that `$acsf_site_name` is getting set correctly, like so:

```
drush @fhl.dev php
>>> global $acsf_site_name
>>> echo $acsf_site_name
fhl
```

However, if I then inspect `global $config`, I see some other config splits being set as expected (environment and sites-directory ones), but nothing for the `fhl` site itself. 

This makes me think that `$acsf_site_name` must not be set at the time that `config.settings.php` is running. I'm not sure why that would be the case though, given that `$acsf_site_name` seems to be getting defined in `factory-hooks/post-sites-php/includes.php`, which runs this:

```php
// Get all the domains that are defined for the current site.
$domains = gardens_data_get_sites_from_file($data['gardens_site_settings']['conf']['acsf_db_name']);
// Get the site's name from the first domain.
global $acsf_site_name;
$acsf_site_name = explode('.', array_keys($domains)[0])[0];
```

Seems like `post-sites-php` must run before `settings.php` does, since the proper `settings.php` file to use is defined by `sites.php`.

Any other thoughts on how I could debug this, or what I could be missing?

**Comment by lcatlett** [comment]
This will be addressed by #3158

**Comment by lcatlett** [comment]
@mikecrittenden see #3172 for the fix.
